layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0002-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch

```diff
@@ -0,0 +1,77 @@
+Index: git/include/environment/ti/mmc.h
+===================================================================
+--- git.orig/include/environment/ti/mmc.h
++++ git/include/environment/ti/mmc.h
+@@ -14,8 +14,8 @@
+ 	"finduuid=part uuid ${boot} ${bootpart} uuid\0" \
+ 	"args_mmc=run finduuid;setenv bootargs console=${console} " \
+ 		"${optargs} " \
+-		"root=PARTUUID=${uuid} rw " \
+-		"rootfstype=${mmcrootfstype}\0" \
++		"${resin_kernel_root}" \
++		"${os_cmdline}"
+ 	"loadbootscript=load mmc ${mmcdev} ${loadaddr} boot.scr\0" \
+ 	"bootscript=echo Running bootscript from mmc${mmcdev} ...; " \
+ 		"source ${loadaddr}\0" \
+@@ -23,8 +23,8 @@
+ 	"importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
+ 		"env import -t ${loadaddr} ${filesize}\0" \
+ 	"loadbootenv=fatload mmc ${mmcdev} ${loadaddr} ${bootenvfile}\0" \
+-	"loadimage=load ${devtype} ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
+-	"loadfdt=load ${devtype} ${bootpart} ${fdtaddr} ${bootdir}/${fdtfile}\0" \
++	"loadimage=load ${devtype} $${resin_dev_index}:${resin_root_part} ${loadaddr} ${bootdir}/${bootfile}\0" \
++	"loadfdt=load ${devtype} ${resin_dev_index}:${resin_root_part} ${fdtaddr} ${bootdir}/${fdtfile}\0" \
+ 	"envboot=mmc dev ${mmcdev}; " \
+ 		"if mmc rescan; then " \
+ 			"echo SD/MMC found on device ${mmcdev};" \
+Index: git/include/environment/ti/mmc.env
+===================================================================
+--- git.orig/include/environment/ti/mmc.env
++++ git/include/environment/ti/mmc.env
+@@ -3,8 +3,7 @@ mmcrootfstype=ext4 rootwait
+ finduuid=part uuid ${boot} ${bootpart} uuid
+ args_mmc=run finduuid;setenv bootargs console=${console}
+ 	${optargs}
+-	root=PARTUUID=${uuid} rw
+-	rootfstype=${mmcrootfstype}
++	${resin_kernel_root}
```

Comment:
I'm failing to boot and I am not entirely sure this value has been set here.

Reply:
if I modify everything but this .env file then PARTUUID is still used. Now if I change it the board fails to boot. My guess is that I need to somehow populate resin_kernel_root before this is called, and I'm failing to see exactly how.

Reply:
@perezmeyer if you have access to the u-boot console, are you able to print the contents of the boot command? Then, run each sub sequence step by step manually, for example:

pri $bootcmd; # check if it contains the correct instructions

# if it does, proceed to
run findfdt; 
setenv resin_kernel_load_addr ${loadaddr}; 
run resin_set_kernel_root;
pri $resin_kernel_root; 

if resin_kernel_root does not look right, run each subcommand in $resin_set_kernel_root;
Otherwise if it does look right, then $resin_kernel_root is not part of the bootcmd

Reply:
Awesome, El si